### PR TITLE
added di with hilt

### PR DESCRIPTION
**Pull request by VivianLi98**
Dependency injection with Dagger Hilt. Added UserRepository class (more methods to come).

### CODE REVIEW

File: app/src/main/java/com/example/voyagerx/repository/LaunchRepository.kt [hunk 1]
```diff
@@ -1,12 +1,20 @@
...
+import dagger.hilt.android.qualifiers.ApplicationContext
+import java.util.*
+import javax.inject.Inject
+import javax.inject.Singleton
 
-
-object LaunchRepository {
+@Singleton
+class LaunchRepository @Inject constructor(
+    private val database: VoyagerXDatabase
```
Comment (joelc1225) [inline app/src/main/java/com/example/voyagerx/repository/LaunchRepository.kt:16]: 🚀

File: app/src/main/java/com/example/voyagerx/repository/LaunchRepository.kt [hunk 2]
```diff
@@ -28,7 +36,7 @@ object LaunchRepository {
         val customLaunches = launches.map{ launch ->
             launch?.let {
             Launch(
-                launch.id!!,
+                launch.id?: uniqueId(),
```
Comment (joelc1225) [inline app/src/main/java/com/example/voyagerx/repository/LaunchRepository.kt:39]: 👍

**Review by joelc1225 — APPROVED** [review]

File: app/src/main/java/com/example/voyagerx/repository/LaunchRepository.kt [hunk 3]
```diff
@@ -42,9 +50,11 @@ object LaunchRepository {
         return customLaunches
     }
 
-//    insert launches into launch table
-//    private fun insertLaunches{
-//
-//    }
+    private fun uniqueId():String = UUID.randomUUID().toString()
```
Comment (Celinarabe) [inline app/src/main/java/com/example/voyagerx/repository/LaunchRepository.kt:53]: Would this ever generate an id that already exists in the table?

File: app/src/main/java/com/example/voyagerx/repository/UserRepository.kt [hunk 4]
```diff
@@ -0,0 +1,20 @@
+package com.example.voyagerx.repository
+
+import com.example.voyagerx.repository.database.VoyagerXDatabase
+import com.example.voyagerx.repository.model.User
+import javax.inject.Inject
+import javax.inject.Singleton
+
+@Singleton
```
Comment (chrsbell) [inline app/src/main/java/com/example/voyagerx/repository/UserRepository.kt:8]: I like the new annotations!

**Review by chrsbell — APPROVED** [review]
Looks good!

**Pull request merged**